interferogram/aoi_merged_track_stitcher.py

```diff
@@ -0,0 +1,355 @@
+        'refbbox': [],
+        'esd_threshold': [],
+        'frame_id': [],
+        'temporal_span': None,
+        'track_number': None,
+        'archive_filename': dataset_id,
+        'dataset_type': 'slc',
+        'tile_layers': ['amplitude', 'displacement'],
+        'latitude_index_min': int(math.floor(env[2] * 10)),
+        'latitude_index_max': int(math.ceil(env[3] * 10)),
+        'parallel_baseline': [],
+        'url': [],
+        'doppler': [],
+        'version': [],
+        'orbit_type': [],
+        'frame_number': None,
+        'bbox': bbox,
+        'ogr_bbox': [[x, y] for y, x in bbox],
+        'orbit_number': [],
+        'input_file': 'ifg_stitch.json',
+        'perpendicular_baseline': [],
+        'orbit_repeat': [],
+        'sensing_stop': endtime,
+        'polarization': [],
+        'scene_count': 0,
+        'beam_id': None,
+        'sensor': [],
+        'look_direction': [],
+        'platform': [],
+        'starting_range': [],
+        'frame_name': [],
+        'tiles': True,
+        'sensing_start': starttime,
+        'beam_mode': [],
+        'image_corners': [], # may keep
+        'prf': [],
+        "sha224sum": hashlib.sha224(str.encode(os.path.basename(met_json_filename))).hexdigest(),
+    }
+
+    # collect values
+    set_params = ('esd_threshold', 'frame_id', 'parallel_baseline', 'doppler', 'orbit_type', 'orbit_number',
+                  'perpendicular_baseline', 'orbit_repeat', 'polarization', 'sensor', 'look_direction', 'platform',
+                  'starting_range', 'beam_mode', 'prf')
+    single_params = ('temporal_span', 'track_number')
+    list_params = ('platform', 'perpendicular_baseline', 'parallel_baseline')
+    mean_params = ('perpendicular_baseline', 'parallel_baseline')
+
+    print("list individual GUNW merged met.json files: {}".format(met_files))
+
+    for i, met_file in enumerate(met_files):
+        with open(met_file) as f:
+            md = json.load(f)
+        for param in set_params:
+            # logger.info("param: {}".format(param))
+            if isinstance(md[param], list):
+                met[param].extend(md[param])
+            else:
+                met[param].append(md[param])
+        if i == 0:
+            for param in single_params:
+                met[param] = md[param]
+        met['scene_count'] += 1
+    for param in set_params:
+        tmp_met = list(set(met[param]))
+        if param in list_params:
+            met[param] = tmp_met
+        else:
+            met[param] = tmp_met[0] if len(tmp_met) == 1 else tmp_met
+    for param in mean_params:
+        met[param] = np.mean(met[param])
+    print('generated met.json object: {}'.format(json.dumps(met, indent=2)))
+
+    with open(met_json_filename, 'w') as f:  # writing the met.json
+        met_json = json.dumps(met, indent=2)
+        f.write(met_json)
+    return True
+
+
+if __name__ == '__main__':
+    VERSION = 'v2.0'  # values needed for the dataset.json file
```

Comment:
The versions of datasets created in this repo is controlled by the `dataset_versions.json` under conf: https://github.com/aria-jpl/ariamh/blob/develop/conf/dataset_versions.json. You can add it there and pull from it to get the version. Example of doing that here: https://github.com/aria-jpl/ariamh/blob/28b0ac9d7fb7242717e7dfe7f247457767e95812/interferogram/stitch_ifgs.py#L24-L33.

Reply:
pushed changes